superset-frontend/src/dashboard/actions/dashboardState.js

```diff
@@ -199,6 +201,28 @@ export function saveDashboardRequest(data, id, saveType) {
       },
     })
       .then(response => {
+        if (isFeatureEnabled(FeatureFlag.DASHBOARD_CROSS_FILTERS)) {
+          const {
+            dashboardInfo: {
+              metadata: { chart_configuration = {} },
+            },
+          } = getState();
+          const chartConfiguration = Object.values(chart_configuration).reduce(
+            (prev, next) => {
+              // If chart removed from dashboard - remove it from metadata
+              if (
+                Object.values(layout).find(
+                  layoutItem => layoutItem?.meta?.chartId === next.id,
+                )
+              ) {
+                return { ...prev, [next.id]: next };
+              }
+              return prev;
+            },
+            {},
+          );
+          dispatch(setChartConfiguration(chartConfiguration));
+        }
```

Comment:
I assume this isn't strictly related to this PR? As this is a pretty heavy PR, could we break this out into a separate PR? (trying to avoid unnecessary bycatches)